### PR TITLE
Add Github action to build Docker image on main and pull requests

### LINKED ISSUE #15: Add Github workflow to auto push docker image to Github Container Registry
We should probably setup a Rocketry docker hub account. I currently just pushed an image to a personal account I made: https://hub.docker.com/u/zachgarcia42. Then we can follow this to auto push: https://docs.github.com/en/actions/use-cases-and-examples/publishing-packages/publishing-docker-images

### PR DESCRIPTION
**Pull request by maxslarsson**
Add Github action to build Docker image on main and pull requests. It publishes these images to the Github Container Registry (which is linked to this repo) so that they can be pulled by anyone. Additionally, images of specific branches that have PRs can be pulled and tested. 

This resolves #15

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#
+name: Create and publish the Ground Station Docker image
+
+# Configures this workflow to run every time a change is pushed to the branch called `release`.
```
Comment (ZachGarcia42) [inline .github/workflows/build.yml:4]: Should be branch called 'main'

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    branches: ['main']
+
+# Defines two custom environment variables for the workflow. These are used for the Container registry domain, and a name for the Docker image that this workflow builds.
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
```
Comment (ZachGarcia42) [inline .github/workflows/build.yml:14]: Can you change the image name to be ground-server, not ground-software? Will be important if we have more docker images in the future.

File: README.md [hunk 3]
```diff
@@ -2,10 +2,12 @@
...
 ## Ground Server
-To start the ground server, run
-```sudo snap install docker```
-```docker pull zachgarcia42/ground-server-image```
-```docker run -it -d -p 80:80 zachgarcia42/ground-server-image```
+
+First, install Docker for your specific computer. Then, to start the ground server run:
+```shell
+docker pull ghcr.io/cornellrocketryteam/ground-software
+docker run -it -d -p 80:80 zachgarcia42/ground-server-image
```
Comment (ZachGarcia42) [inline README.md:9]: Should update the run line to be ghcr.io/cornellrocketryteam/ground-software. Or ghcr.io/cornellrocketryteam/ground-server if we change the image name.

**Review by ZachGarcia42 — CHANGES_REQUESTED** [review]
Thanks for this work! This will really help the team

File: README.md [hunk 4]
```diff
@@ -2,10 +2,12 @@
...
 ## Ground Server
-To start the ground server, run
-```sudo snap install docker```
-```docker pull zachgarcia42/ground-server-image```
-```docker run -it -d -p 80:80 zachgarcia42/ground-server-image```
+
+First, install Docker for your specific computer. Then, to start the ground server run:
+```shell
+docker pull ghcr.io/cornellrocketryteam/ground-server
+docker run -it -d -p 80:80 ghcr.io/cornellrocketryteam/ground-server
```
Comment (ZachGarcia42) [inline README.md:9]: Should this be give.io/ground-server? The build/push rule defines env.Registry/env.Image_Name which wouldn't include "cornellrocketryteam", right?

Comment (maxslarsson) [reply README.md:9]: So I'm pretty sure it actually automatically scopes the image under the user, so in this case cornellrocketryteam. Before, the env.Image_Nae was actually just ground-software (according to [this](https://stackoverflow.com/questions/70700244/in-a-github-actions-workflow-how-can-you-determine-the-organziation-where-the-c) stack overflow post). So it adds the cornellrocketryteam automatically.

Comment (ZachGarcia42) [reply README.md:9]: Cool, thanks for clarifying!

**Review by ZachGarcia42 — APPROVED** [review]

**Pull request merged**